sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppImpl.java

```diff
@@ -160,10 +160,21 @@ public boolean hasConfigurationServiceBinding() {
             return false;
         }
         return addonConfigs.get(Constants.APPLICATION_CONFIGURATION_SERVICE_KEY) != null
-            && Objects.equals(
-            addonConfigs.get(Constants.APPLICATION_CONFIGURATION_SERVICE_KEY).get(Constants.BINDING_RESOURCE_ID),
-            configurationService.id()
-        );
+            && configurationService.id().equalsIgnoreCase((String) addonConfigs.get(Constants.APPLICATION_CONFIGURATION_SERVICE_KEY).get(Constants.BINDING_RESOURCE_ID));
+    }
+
+    @Override
+    public boolean hasServiceRegistryBinding() {
+        Map<String, Map<String, Object>> addonConfigs = innerModel().properties().addonConfigs();
+        if (addonConfigs == null) {
+            return false;
+        }
+        SpringServiceRegistry serviceRegistry = parent().getDefaultServiceRegistry();
+        if (serviceRegistry == null) {
+            return false;
+        }
+        return addonConfigs.get(Constants.SERVICE_REGISTRY_KEY) != null
+            && serviceRegistry.id().equalsIgnoreCase((String) addonConfigs.get(Constants.SERVICE_REGISTRY_KEY).get(Constants.BINDING_RESOURCE_ID));
```

Comment:
is `serviceRegistry.id()` guaranteed not null?

Reply:
`serviceRegistry` is from remote request and `id` is from its `innerModel().id()`. If service registry itself is not null, its id can't be null I assume.